### PR TITLE
[VMware] Implementation of user selectable guesttype and NIC interface type

### PR DESCRIPTION
**Pull request by MarcGrimme**
This is my latest changes to foreman-stable-1.1 branch that extend the vmware compute-resource with user selectable guest types and nic interfacetypes.
Also the GUI now shows the name and mac of the NIC to be better identified.

Hope this helps somehow.

Regards Marc.

### CODE REVIEW

**Comment by ohadlevy** [comment]
hey, this is great, but it seems it got mixed with other commits, can you rebase?

**Comment by MarcGrimme** [comment]
Hmm, I've rebased my branch vmware-guesttype to your develop branch and commited the changes.
Hopefully those should be seen here soon.

Nevertheless foreman is not running based on develop.
I get a 
"Operation FAILED: undefined method `type_changed?' for #Host::Managed:0x7f1f534b5018"
when trying to create a new host.

Did I miss a commit?

Thanks Marc.

**Comment by MarcGrimme** [comment]
Ok, did a dbmigrate and now everything looks fine.

But I still don't see my rebase in this pull request.
Shall I create a new one?

Thanks Marc.

**Comment by GregSutcliffe** [comment]
After a rebase, you normally need to push with -f to github to force-update the history changes - did you do that?

**Comment by MarcGrimme** [comment]
Yes I think so. My rebase changes are at github.

**Comment by ohadlevy** [comment]
@grimme-atix-de after rebase only your commits should be visible above....

**Comment by ohadlevy** [comment]
@grimme-atix-de looking great! I'll try to play a bit later on, as we are currently in the mid of upgrading to rails3.2

**Comment by ohadlevy** [comment]
[test]

**Comment by ohadlevy** [comment]
sadly my vsphere setup died, once recovered I'll test asap

File: app/views/compute_resources_vms/form/vmware/_network.html.erb [hunk 1]
```diff
@@ -2,7 +2,18 @@
   <% if (networks = compute_resource.networks).any? -%>
     <%= selectable_f f, :network, networks, { },
                      :class       => "span2",
-                     :label       => "Network",
-                     :help_inline => remove_child_link("X", f, { :method => :'_delete', :title => 'remove network interface', :class => 'label label-important' }) %>
+                     :label       => "Network" %>
+  <% end -%>
+  <!-- change mac is not tested so disable for the first -->
+  <%= text_f f, :mac, :hidden => newmachine, :disabled => true unless not !newmachine and f.object.respond_to? :mac%>
```
Comment (ohadlevy) [inline app/views/compute_resources_vms/form/vmware/_network.html.erb:8]: you simply be !(expression)

**Comment by ohadlevy** [comment]
I've tried to merge it, but there a lot of merge conflicts, please rebase on top of current develop + squash your commits?

thanks!

**Comment by MarcGrimme** [comment]
I merged it to current develop. Is it better now?

**Comment by ohadlevy** [comment]
no :(, it should appear with only your comits.

try git rebase -m origin/develop ?
then you could also do rebase -i to squash all remaining commits?

**Comment by MarcGrimme** [comment]
Hopefully better now. I tried to do what you said (git rebase -m .., git rebase -i).
What do you think now?

**Comment by ohadlevy** [comment]
I'm not sure.. Im still seeing lots of commits... any chance you are pushing the wrong branch?

**Comment by MarcGrimme** [comment]
So I've remerged all changes into one commit in my branch. Somehow there is still the release 1.1-1 commit from you. Hope this does not hurt.
Let me know what you think.

Keep in mind, this will only work with https://github.com/fog/fog/pull/1582 applied to fog.

File: app/assets/javascripts/compute_resource.js [hunk 2]
```diff
@@ -77,6 +77,43 @@ function add_network_interface(item){
...
+  $('#hwp_indicator').show();
+  $('#interfacetypes').children('option').remove()
+  $.ajax({
+      type:'post',
+      url: url,
+      data:'hwp_id=' + servertype,
+      success: function(result){
+        $.each(result.interfacetypes, function() {add_network_interfacetype(this);});
+        $('#network_interfaces').children('.fields').remove();
+//        $.each(result.interfaces, function() {add_network_interface(this);});
```
Comment (ohadlevy) [inline app/assets/javascripts/compute_resource.js:104]: why commented out?

Comment (MarcGrimme) [inline app/assets/javascripts/compute_resource.js:104]: As far a I recall I couldn't see a reason for this. Besides the function seems to be working without that code.

Do you recall what's the reason for this line?

**Comment by domcleal** [comment]
Thanks for the PR, but we've let this grow out of date with respect to current 'develop'.  Some of this has been implemented (e.g. the NIC selection), but the guest types would be a great addition.

If you'd like to rebase it and fit it into the current structure, we'd very much like to include it - I now have some test capability for VMware too.  For now though, I'm going to close this as it's diverged too much.

**Pull request closed without merging**